src/request.rs

```diff
@@ -0,0 +1,118 @@
+use napi::bindgen_prelude::{BigInt, Buffer};
+use scylla::frame::{
+    response::result::CqlValue,
+    value::{Counter, CqlTimeuuid},
+};
+
+use crate::{
+    types::uuid::UuidWrapper,
+    utils::{bigint_to_i64, js_error},
+};
+
+#[napi]
+pub struct QueryParameterWrapper {
+    parameter: CqlValue,
+}
+
+#[napi]
+impl QueryParameterWrapper {
+    #[napi]
+    pub fn from_ascii(val: String) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Ascii(val),
+        }
+    }
+
+    #[napi]
+    pub fn from_boolean(val: bool) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Boolean(val),
+        }
+    }
+
+    #[napi]
+    pub fn from_blob(val: Buffer) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Blob(val.to_vec()),
+        }
+    }
+
+    #[napi]
+    pub fn from_counter(val: BigInt) -> napi::Result<QueryParameterWrapper> {
+        Ok(QueryParameterWrapper {
+            parameter: CqlValue::Counter(Counter(bigint_to_i64(
+                val,
+                "Value casted into counter type shouldn't overflow i64",
+            )?)),
+        })
+    }
+
+    #[napi]
+    pub fn from_double(val: f64) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Double(val),
+        }
+    }
+
+    #[napi]
+    pub fn from_float(val: f64) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Float(val as f32),
+        }
+    }
+
+    #[napi]
+    pub fn from_int(val: i32) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Int(val),
+        }
+    }
+
+    #[napi]
+    pub fn from_text(val: String) -> QueryParameterWrapper {
+        QueryParameterWrapper {
+            parameter: CqlValue::Ascii(val),
+        }
+    }
+
+    #[napi]
+    pub fn from_set(val: Vec<&QueryParameterWrapper>) -> QueryParameterWrapper {
```

Comment:
This seems a very weird type for a set to me. Why have a `Vec` of references at all?
Perhaps this function should accept `impl Iterator<Item = &QueryParameterWrapper>`?

Reply:
Also, we don't seem to have list nor map supported here, only set. Why so?

Reply:
1. This functions is exposed to napi, so I don't think iterators will work here.

2. Those types are implemented here: https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/pull/95